ts/components/session/SessionNicknameDialog.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { useState } from 'react';
+import { ConversationController } from '../../session/conversations/ConversationController';
+import { SessionModal } from './SessionModal';
+import { SessionButton, SessionButtonColor } from './SessionButton';
+import { SessionHtmlRenderer } from './SessionHTMLRenderer';
+import { SessionIcon, SessionIconSize, SessionIconType } from './icon';
+import { DefaultTheme, withTheme } from 'styled-components';
+
+type Props = {
+  message: string;
+  messageSub: string;
+  title: string;
+  placeholder?: string;
+  onOk?: any;
+  onClose?: any;
+  onClickOk: any;
+  onClickClose: any;
+  okText?: string;
+  cancelText?: string;
+  hideCancel: boolean;
+  okTheme: SessionButtonColor;
+  closeTheme: SessionButtonColor;
+  sessionIcon?: SessionIconType;
+  iconSize?: SessionIconSize;
+  theme: DefaultTheme;
+  convoId?: string;
```

Comment:
no need for all those args. You can just keep the one you need you actually need to be dynamic.

like the okTheme is always `SessionButtonColor.Primary` for this dialog so no need to pass it in the props, same for the okText of stuff like this.